hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/streams/RecordAssertions.java

```diff
@@ -85,12 +86,20 @@ public static void triggerAndCloseAtLeastOneFile(final HapiSpec spec) throws Int
     }
 
     public static void triggerAndCloseAtLeastOneFileIfNotInterrupted(final HapiSpec spec) {
-        try {
+        doIfNotInterrupted(() -> {
             RecordAssertions.triggerAndCloseAtLeastOneFile(spec);
             LOG.info("Sleeping a bit to give the record stream a chance to close");
             Thread.sleep(BLOCK_CREATION_SLEEP_PERIOD.toMillis());
-        } catch (final InterruptedException ignore) {
+        });
+    }
+
+    public static void doIfNotInterrupted(@NonNull final InterruptibleRunnable runnable) {
+        requireNonNull(runnable);
+        try {
+            runnable.run();
+        } catch (InterruptedException e) {
             Thread.currentThread().interrupt();
+            throw new IllegalStateException(e);
```

Comment:
interesting!  I've often wondered about that Java recommendation about always rethrowing `InterruptedException` ...

Reply:
Sonar likes it. 🤷